exercises/bank-account/canonical-data.json

```diff
@@ -0,0 +1,303 @@
+      "input": {
+        "operations": [
+          {
+            "operation": "open"
+          },
+          {
+            "operation": "amount",
+            "expected": 0
+          }
+        ]
+      },
+      "expected": {}
+    },
+    {
+      "uuid": "e88d4ec3-c6bf-4752-8e59-5046c44e3ba7",
+      "description": "Can deposit money",
+      "property": "bankAccount",
+      "input": {
+        "operations": [
+          {
+            "operation": "open"
+          },
+          {
+            "operation": "deposit",
+            "value": 100
+          },
+          {
+            "operation": "amount",
+            "expected": 100
+          }
+        ]
+      },
+      "expected": {}
+    },
+    {
+      "uuid": "3d9147d4-63f4-4844-8d2b-1fee2e9a2a0d",
+      "description": "Can deposit money sequentially",
+      "property": "bankAccount",
+      "input": {
+        "operations": [
+          {
+            "operation": "open"
+          },
+          {
+            "operation": "deposit",
+            "value": 100
+          },
+          {
+            "operation": "deposit",
+            "value": 50
+          },
+          {
+            "operation": "amount",
+            "expected": 150
+          }
+        ]
+      },
+      "expected": {}
+    },
+    {
+      "uuid": "08f1af07-27ae-4b38-aa19-770bde558064",
+      "description": "Can withdraw money",
+      "property": "bankAccount",
+      "input": {
+        "operations": [
+          {
+            "operation": "open"
+          },
+          {
+            "operation": "deposit",
+            "value": 100
+          },
+          {
+            "operation": "withdraw",
+            "value": 50
+          },
+          {
+            "operation": "amount",
+            "expected": 50
+          }
+        ]
+      },
+      "expected": {}
+    },
+    {
+      "uuid": "6f6d242f-8c31-4ac6-8995-a90d42cad59f",
+      "description": "Can withdraw money sequentially",
+      "property": "bankAccount",
+      "input": {
+        "operations": [
+          {
+            "operation": "open"
+          },
+          {
+            "operation": "deposit",
+            "value": 100
+          },
+          {
+            "operation": "withdraw",
+            "value": 80
+          },
+          {
+            "operation": "withdraw",
+            "value": 20
+          },
+          {
+            "operation": "amount",
+            "expected": 0
+          }
+        ]
+      },
+      "expected": {}
+    },
+    {
+      "uuid": "f9facfaa-d824-486e-8381-48832c4bbffd",
+      "description": "Checking balance of closed account throws error",
+      "property": "bankAccount",
+      "input": {
+        "operations": [
+          {
+            "operation": "open"
+          },
+          {
+            "operation": "close"
+          },
+          {
+            "operation": "amount"
+          }
+        ]
+      },
+      "expected": { "error": "account not open" }
```

Comment:
It might be a good idea to make this convention clear though - the comments would say something like "if an error is expected, the last operation is the one that caused it; all operations other than the last one should succeed"

Reply:
I will see if other reviewers feels like there needs to be a structure change if not I will add that as a comment that it is the last one that should always be the action which will raise error